test/libsolidity/syntaxTests/constantEvaluator/overflow.sol

```diff
@@ -0,0 +1,9 @@
+contract C {
+    uint8 constant a = 255;
+    uint16 constant b = a + 2;
+    function f() public pure {
+        uint[b] memory x;
+    }
+}
+// ----
+// TypeError 2643: (65-70): Arithmetic error when computing constant value.
```

Comment:
The previous behavior was to compute the whole expression with pure integers, so the array was allocated with length 257. We should document this.

Reply:
So this error happens already at `a + 2`, right?

Reply:
Not really. If you don't use it as an array length, you will only get a runtime panic.